### PR TITLE
dlopen: fix bpf symbol loading check and log at debug level when loading a library

### PR DESCRIPTION
**Pull request by bluca**

### CODE REVIEW

**Comment by bluca** [comment]
Trying to switch bpf loading to dlopen_many_sym_or_warn() but it's segfaulting later when loading additional symbols, which is weird:

```
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7fd49c4 in _dl_lookup_symbol_x (undef_name=0x7ffff7e61c24 "bpf_link__destroy", undef_map=<optimized out>, 
    ref=0x7fffffffd738, symbol_scope=<optimized out>, version=0x0, type_class=0, flags=2, skip_map=0x0)
    at ./elf/dl-lookup.c:771
771	./elf/dl-lookup.c: No such file or directory.
(gdb) bt
#0  0x00007ffff7fd49c4 in _dl_lookup_symbol_x (undef_name=0x7ffff7e61c24 "bpf_link__destroy", 
    undef_map=<optimized out>, ref=0x7fffffffd738, symbol_scope=<optimized out>, version=0x0, type_class=0, flags=2, 
    skip_map=0x0) at ./elf/dl-lookup.c:771
#1  0x00007ffff796e810 in do_sym (handle=<optimized out>, name=0x7ffff7e61c24 "bpf_link__destroy", 
    who=0x7ffff7cd4868 <dlsym_many_or_warnv+117>, vers=vers@entry=0x0, flags=flags@entry=2) at ./elf/dl-sym.c:153
#2  0x00007ffff796ebdd in _dl_sym (handle=<optimized out>, name=<optimized out>, who=<optimized out>)
    at ./elf/dl-sym.c:195
#3  0x00007ffff78a4524 in dlsym_doit (a=a@entry=0x7fffffffd9a0) at ./dlfcn/dlsym.c:40
#4  0x00007ffff796dfca in __GI__dl_catch_exception (exception=exception@entry=0x7fffffffd900, 
    operate=<optimized out>, args=<optimized out>) at ./elf/dl-error-skeleton.c:208
#5  0x00007ffff796e07f in __GI__dl_catch_error (objname=0x7fffffffd958, errstring=0x7fffffffd960, 
    mallocedp=0x7fffffffd957, operate=<optimized out>, args=<optimized out>) at ./elf/dl-error-skeleton.c:227
#6  0x00007ffff78a3f27 in _dlerror_run (operate=operate@entry=0x7ffff78a4510 <dlsym_doit>, 
    args=args@entry=0x7fffffffd9a0) at ./dlfcn/dlerror.c:138
#7  0x00007ffff78a45b5 in dlsym_implementation (dl_caller=<optimized out>, name=<optimized out>, 
    handle=<optimized out>) at ./dlfcn/dlsym.c:54
#8  ___dlsym (handle=<optimized out>, name=<optimized out>) at ./dlfcn/dlsym.c:68
#9  0x00007ffff7cd4868 in dlsym_many_or_warnv (dl=0x100000000, log_level=7, ap=0x7fffffffda70)
    at ../src/basic/dlfcn-util.c:18
#10 0x00007ffff7cd4a50 in dlsym_many_or_warn_sentinel (dl=0x100000000, log_level=7) at ../src/basic/dlfcn-util.c:34
#11 0x00007ffff7aa678e in dlopen_bpf () at ../src/shared/bpf-dlopen.c:97
#12 0x0000555555555a28 in run (argc=1, argv=0x7fffffffdde8) at ../src/test/test-dlopen-so.c:59
#13 0x0000555555555ca1 in main (argc=1, argv=0x7fffffffdde8) at ../src/test/test-dlopen-so.c:94
```

File: src/shared/idn-util.c [hunk 1]
```diff
@@ -42,23 +42,20 @@ int dlopen_idn(void) {
...
                         DLSYM_ARG(stringprep_ucs4_to_utf8),
                         DLSYM_ARG(stringprep_utf8_to_ucs4));
+        if (r == -EOPNOTSUPP) {
+                /* libidn broke ABI in 1.34, but not in a way we care about (a new field got added to an
+                 * open-coded struct we do not use), hence support both versions. */
+                r = dlopen_many_sym_or_warn(&dl, "libidn.so.11", LOG_DEBUG,
+                                DLSYM_ARG(idna_to_ascii_4i),
+                                DLSYM_ARG(idna_to_unicode_44i),
+                                DLSYM_ARG(stringprep_ucs4_to_utf8),
+                                DLSYM_ARG(stringprep_utf8_to_ucs4));
```
Comment (poettering) [inline src/shared/idn-util.c]: why this change? duplicating the list of functios doesn't really sound like an imprvement?

File: src/shared/bpf-dlopen.c [hunk 2]
```diff
@@ -100,7 +101,8 @@ int dlopen_bpf(void) {
                                 DLSYM_ARG_FORCE(libbpf_probe_bpf_prog_type)
 #endif
                 );
-        }
+        if (r < 0)
+                return log_debug_errno(r, "Failed to load symbols from libbpf: %m");
```
Comment (poettering) [inline src/shared/bpf-dlopen.c]: no need for a log msg here. the functions already logged at LOG_DEBUG level, you even specified the intended log level

File: src/shared/idn-util.c (hunk 1 above)
Comment (bluca) [reply src/shared/idn-util.c]: So that the common dlopen helper is used, with the logging and all that

Comment (poettering) [reply src/shared/idn-util.c]: dlopen_many_sym_or_warn() is not an interesting helper. it just combines dlopen() and dlsym_many_or_warn() into one, nothing else. The interesting bits are in dlsym_many_or_warn() and that one the old cod was already using.

Comment (bluca) [reply src/shared/idn-util.c]: ok
